src/menu/menu.c

```diff
@@ -148,21 +150,76 @@ item_create(struct menu *menu, const char *text)
 	wlr_scene_node_set_position(menuitem->selected.text, x, y);
 
 	/* Position the item in relation to its menu */
-	int item_count = wl_list_length(&menu->menuitems);
-	y = item_count * menu->item_height;
+	y = 0;
+	struct menuitem *item;
+	wl_list_for_each_reverse(item, &menu->menuitems, link) {
+		y += item->height;
+	}
 	wlr_scene_node_set_position(&menuitem->tree->node, 0, y);
 
 	/* Hide selected state */
 	wlr_scene_node_set_enabled(&menuitem->selected.tree->node, false);
 
 	/* Update menu extents */
-	menu->size.height = (item_count + 1) * menu->item_height;
+	menu->size.height = y + menuitem->height;
 
 	wl_list_insert(&menu->menuitems, &menuitem->link);
 	wl_list_init(&menuitem->actions);
 	return menuitem;
 }
 
+static struct menuitem *
+separator_create(struct menu *menu, const char *label)
+{
+	int menu_separator_width = 1;
+	int menu_separator_padding_width = 10;
+	int menu_separator_padding_height = 2;
+	float menu_separator_color[4] = { 0.5, 0.5, 0.5, 1.0 };
```

Comment:
Does Openbox provide theme variables for all of these settings?

Reply:
Yep, same named, just dot instead of underscore…. two final cat A items